src/sparseml/optim/helpers.py

```diff
@@ -207,15 +209,52 @@ def update_recipe_variables(recipe_yaml_str: str, variables: Dict[str, Any]) ->
         # yaml string does not create a dict, return original string
         return recipe_yaml_str
 
-    for key in variables:
-        if key not in container:
+    if check_if_staged_recipe(container):
+        key_found = False
```

Comment:
should this be at the scope of the loop below so we can raise an error on any missing key?